src/month.rs

```diff
@@ -8,29 +8,28 @@ use crate::OutOfRange;
 /// The month of the year.
 ///
 /// This enum is just a convenience implementation.
-/// The month in dates created by DateLike objects does not return this enum.
+/// The month in dates created by `Datelike` objects does not return this enum.
 ///
-/// It is possible to convert from a date to a month independently
+/// It is possible to convert from a date to a month using `try_from`:
 /// ```
 /// use chrono::prelude::*;
 /// use chrono::Month;
 ///
 /// let date = Utc.at_ymd_and_hms(2019, 10, 28, 9, 10, 11).unwrap();
 /// // `2019-10-28T09:10:11Z`
-/// let month = Month::try_from(u8::try_from(date.month()).unwrap()).ok();
-/// assert_eq!(month, Some(Month::October))
+/// let month = Month::try_from(date.month() as u8).unwrap();
+/// assert_eq!(month, Month::October)
 /// ```
-/// Or from a Month to an integer usable by dates
+/// Or from a `Month` to an integer with simple casting:
 /// ```
 /// # use chrono::prelude::*;
 /// # use chrono::Month;
 /// let month = Month::January;
-/// let dt = Utc.at_ymd_and_hms(2019, month.number_from_month(), 28, 9, 10, 11).unwrap();
+/// let dt = Utc.at_ymd_and_hms(2019, month as u32, 28, 9, 10, 11).unwrap();
```

Comment:
Should we change `at_ymd_and_hms()` (and other places) to take `u8` for `month`?

Reply:
I don't know. A number of methods could take smaller arguments, such as `NaiveTime::from_hms`. Is there an advantage for users in taking a smaller type?

Reply:
Smaller types would probably allow more optimization potential.

Reply:
I'll investigate and open an issue or PR. It does mean users have to do potentially fallible casts more often, so I'm not sure it is better for egonomics.